packages/headless-react/typedoc/documents/hooks.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Controller hooks
+---
+
+# Hooks
+
+This package provides React hooks for controllers defined in the Engine Definition. Hooks allow you to access controller state and methods easily within your React components.
+
+## Usage
+
+1. Create an engine configuration
+
+Define the controllers you need in your engine configuration. This example includes `Summary` and `ProductList` controllers:
+
+```
+const config : CommerceEngineDefinitionOptions {
+    configuration: {
+    ...getSampleCommerceEngineConfiguration(),
+  },
+  controllers: {
+    summary: defineSummary(),
+    productList: defineProductList(),
+  }
+}
+```
+
+2. Define the engine
+
+Use `defineCommerceEngine` to define the engine with your configuration:
+
+```
+export const engineDefinition = defineCommerceEngine(engineConfig);
+```
+
+3. Export hooks
+
+Extract hooks for each controller from the engine definition.
+
+```
+export const {
+  useSummary,
+  useProductList,
+} = engineDefinition.controllers;
+```
+
+4. Use Hooks in Components
+
+Access controller methods and state in your components through the hooks.
+The controller methods and state attributes exposed through controller hooks are the same as the ones exposed by the controllers.
+For details, see the relevant controller [reference documentation](../../modules/SSR_Commerce.index.html).
+
+```
+export default function ProductList() {
+  const {state, methods} = useProductList();
+  const {state: cartState, methods: cartMethods} = useCart();
```

Comment:
For the sake of clarity, I think you should define the cart and export the hook in the examples above. Perhaps instead of the summary in oder to keep things concise.

Reply:
good catch